### ISSUE TITLE
fish crash on certain autocomplete like `fi` and tab

### ISSUE THREAD

**Issue by christianrondeau** [termux/termux-packages#909]
Version 2.5.0 of fish crashes when I type `fi`, then <kbd>Tab</kbd>. It displays :

    fisrc/pager.cpp:161: line_t pager_t::completion_print_item(const wcstring &, const pager_t::comp_t *, size_t, size_t, size_t, bool, bool, page_rendering_t *) const: assertion "desc_remaining >= 2" failed

It freezes for a few seconds, then:

    Aborted

**Comment by agusdallalba** [comment]
This happens to me too. Also I get a segfault when I type `t`. I'm running termux on a Nexus 5x (aarch64).

I ran it with strace to see if I could get something more meaningful.

```
rt_sigprocmask(SIG_BLOCK, [CHLD], [], 8) = 0
read(0, "t", 1)                         = 1
rt_sigprocmask(SIG_SETMASK, [], [CHLD], 8) = 0
pselect6(1, [0], NULL, NULL, {tv_sec=0, tv_nsec=0}, NULL) = 0 (Timeout)
rt_sigprocmask(SIG_BLOCK, ~[], [], 8)   = 0
mmap(NULL, 1036288, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_NORESERVE, -1, 0) = 0x7d1774a000
mprotect(0x7d1774a000, 4096, PROT_NONE) = 0
prctl(PR_SET_VMA, PR_SET_VMA_ANON_NAME, 0x7d1774a000, 4096, "thread stack guard page") = 0
mmap(NULL, 20480, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7d1822c000
prctl(PR_SET_VMA, PR_SET_VMA_ANON_NAME, 0x7d1822c000, 20480, "bionic TLS guard page") = 0
mprotect(0x7d1822d000, 12288, PROT_READ|PROT_WRITE) = 0
prctl(PR_SET_VMA, PR_SET_VMA_ANON_NAME, 0x7d1822d000, 12288, "bionic TLS") = 0
clone(child_stack=0x7d178464e0, flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, parent_tidptr=0x7d17846500, tls=0x7d17846588, child_tidptr=0x7d17846500) = 25215
futex(0x7d17846570, FUTEX_WAKE_PRIVATE, 1) = 1
rt_sigprocmask(SIG_SETMASK, [], ~[KILL STOP], 8) = 0
rt_sigprocmask(SIG_BLOCK, ~[], [], 8)   = 0
mmap(NULL, 1036288, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_NORESERVE, -1, 0) = 0x7d1764d000
mprotect(0x7d1764d000, 4096, PROT_NONE) = ?
+++ killed by SIGSEGV +++
Segmentation fault
```

It doesn't mean much to me.

**Comment by agusdallalba** [comment]
Since latest upgrade (2.5.0-1 → 2.6.0):

- `fi<Tab>`: segfault

- `t`: works

**Comment by fornwall** [comment]
Do you still have this problem? I'm unable to reproduce it.

Those still having the problem, could you state Android version and phone model to see if it could be depending on that?

**Comment by christianrondeau** [comment]
> fi<E> fish: src/pager.cpp:181: failed assertion: desc_remaining >= 2rmine file type)
    <E> fish: Sorry, but your system does not support backtraces files in a directory hi…)
    findfs  (Find a filesystem by label or UU…)
    [Process completed (signal 6) - press Enter]

Android version:
7.0
Device manufacturer:
samsung
Device model:
SM-G935W8

**Comment by christianrondeau** [comment]
Looks like the crash "improved" :P

**Comment by agusdallalba** [comment]
What triggers the bug, and what kind of message you get from it changes on every fish version. Right now fi<tab> aborts in the same way as the previous comment.

I'm running Android 8.0 on a Nexus 5x. I notice @christianrondeau's CPU is also 64-bit. That might have something to do with this.

**Comment by fornwall** [comment]
This seems to be fixed in the newly updated `2.7.0` version of fish!
